src/main/java/org/opensrp/web/config/security/OAuth2SecurityConfig.java

```diff
@@ -83,7 +87,18 @@ public DefaultTokenServices tokenServices() {
 	
 	@Bean
 	public JdbcTokenStore tokenStore() {
-		return new JdbcTokenStore(dataSource);
+		final JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+		final AuthenticationKeyGenerator authenticationKeyGenerator = new DefaultAuthenticationKeyGenerator();
+		return new JdbcTokenStore(dataSource) {
+			
+			@Override
+			public void storeAccessToken(final OAuth2AccessToken token, final OAuth2Authentication authentication) {
+				final String key = authenticationKeyGenerator.extractKey(authentication);
+				jdbcTemplate.update("delete from oauth_access_token where authentication_id = ?", key);
```

Comment:
Check for null on `key`.

Reply:
Log info when `jdbcTemplate.update("delete from oauth_access_token where authentication_id = ?", key);` eventually finds a duplicate and deletes it.